src/config/oranda_config.rs

```diff
@@ -30,23 +40,39 @@ pub struct MdBookConfig {
     pub theme: Option<bool>,
 }
 
+impl ApplyLayer for MdBookConfig {
+    fn apply_layer(&mut self, layer: Self) {
+        self.path.apply_opt(layer.path);
+        self.theme.apply_opt(layer.theme)
+    }
+}
+
 /// Config related to styling your page
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Default, Deserialize)]
 pub struct StyleConfig {
-    pub theme: Theme,
-    pub syntax_theme: SyntaxTheme,
+    pub theme: Option<Theme>,
+    pub syntax_theme: Option<SyntaxTheme>,
+    #[serde(default)]
     pub additional_css: Vec<String>,
     pub oranda_css_version: Option<String>,
 }
 
-impl Default for StyleConfig {
-    fn default() -> Self {
-        StyleConfig {
-            theme: Theme::Dark,
-            additional_css: vec![],
-            syntax_theme: SyntaxTheme::MaterialTheme,
-            oranda_css_version: None,
-        }
+impl ApplyLayer for StyleConfig {
+    fn apply_layer(&mut self, layer: Self) {
+        self.theme.apply_opt(layer.theme);
+        self.syntax_theme.apply_opt(layer.syntax_theme);
+        self.oranda_css_version.apply_opt(layer.oranda_css_version);
+        self.additional_css.extend(layer.additional_css);
+    }
+}
+impl StyleConfig {
+    /// Get the theme
+    pub fn theme(&self) -> Theme {
```

Comment:
why are we adding functions here instead of using the default values as values?

Reply:
This is the crux of a discussion I brought up with @shadows-withal last week.  Some values in Theme should be Optional in OrandaConfig, but aren't Optional in the final Config. This raises a question: "Do we want the final Config to use the same struct for things like StyleConfig that OrandaConfig uses?" (Before my PR the answer was "yes" and after my PR it remains "yes")

* If no, then we can just have different Optionality and everything is clean
* If yes, then users of the final Config need to deal with the fact that the value is Optional, even though it isn't. I chose to deal with this fact by removing the default value from the Default impl, and moving it to a method that wraps them up in a method that returns the default value.